### PR TITLE
Cleanup: Split fontcache implementations.

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

fontcache.cpp contains 4 font cache related classes, 

1. Base `FontCache`.
2. The sprite-based `SpriteFontCache`.
3. `TrueTypeFontCache` base font cache for platform specific implementations.
4. `FreeTypeFontCache` font cache.

This is left over from the original system which was then expanded. There is therefore a bit of a mishmash of similar looking classes all smashed together.

The platform specific `Win32FontCache` and `CoreTextFontCache` implementations are already in separate files.

Additionally some parts of the fontcache system reference freetype even though they have nothing to do with it.

And the GetUnicodeGlyph method should be internal to the `SpriteFontCache` but it is exposed throughout the font cache system.

## Description

The font cache implementations have been split into separate files into a fontcache subdirectory.

`FreeTypeSettings` has been renamed `FontCacheSettings`.

GetUnicodeGlyph has been removed from FontCache and made private in SpriteFontCache.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/debug.cpp [hunk 1]
```diff
@@ -72,7 +72,7 @@ struct DebugLevel {
 	DEBUG_LEVEL(oldloader),
 	DEBUG_LEVEL(npf),
 	DEBUG_LEVEL(yapf),
-	DEBUG_LEVEL(freetype),
+	DEBUG_LEVEL(fontcache),
```
Comment (LordAro) [inline src/debug.cpp:75]: I suppose technically this makes it a 'change' commit

File: src/fontcache/freetypefontcache.cpp [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+#endif /* WITH_FREETYPE */
+
+
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(WITH_FONTCONFIG) && !defined(WITH_COCOA)
+
+#ifdef WITH_FREETYPE
+FT_Error GetFontByFaceName(const char *font_name, FT_Face *face) { return FT_Err_Cannot_Open_Resource; }
+#endif /* WITH_FREETYPE */
+
+#endif /* !defined(_WIN32) && !defined(__APPLE__) && !defined(WITH_FONTCONFIG) && !defined(WITH_COCOA) */
```
Comment (LordAro) [inline src/fontcache/freetypefontcache.cpp]: Why separate this from the main WITH_FREETYPE block?

File: src/fontcache_internal.h [hunk 3]
```diff
@@ -19,59 +19,4 @@ static const int MAX_FONT_SIZE = 72; ///< Maximum font size.
 static const byte FACE_COLOUR = 1;
 static const byte SHADOW_COLOUR = 2;
```
Comment (LordAro) [inline src/fontcache_internal.h]: fontcache_internal is now left with just 3 constants. Can they go somewhere else?

**Review by LordAro — COMMENTED** [review]
<3

File: src/fontcache/freetypefontcache.cpp (hunk 2 above)
Comment (PeterN) [reply src/fontcache/freetypefontcache.cpp]: I remembered cleaning this bit up, but that must have been in a slightly different iteration!

File: src/fontcache_internal.h (hunk 3 above)
Comment (PeterN) [reply src/fontcache_internal.h]: I think so, They are all only used by TrueTypeFontCache implementations, but the two colour definitions could be considered generic enough to just live in fontcache.h.

I think for now I should simply move them along with the other TrueType bits.

**Review by LordAro — APPROVED** [review]

**Pull request merged**